src/consumers.ts

```diff
@@ -94,7 +103,11 @@ class DiagnosticsPipeline implements IPipeline<Vulnerability[]>
                             newText: aggVulnerability.recommendedVersion
                         }];
                         // We will have line|start as key instead of message
-                        codeActionsMap[aggDiagnostic.range.start.line + "|" + aggDiagnostic.range.start.character] = codeAction;
+                        codeActionsMap[aggDiagnostic.range.start.line + "|" + aggDiagnostic.range.start.character] = {fix: codeAction, report: fullStackReportAction};
```

Comment:
@dvandra Instead of so many changes, Can we set some Diagnostic code and decide to provide full stack CodeAction based on that from onCodeAction event?

Reply:
@arajkumar Are you suggesting to do this using diagnostic.code?

Reply:
Yeah, another option is to check the `source` of the Diagnostic. if source == `Dependency Analytics Plugin [Powered by Snyk]` then generate CodeAction for fullstack report.